components/outputs/plot-matplotlib.qmd

```diff
@@ -0,0 +1,93 @@
+      code: |4
+          import matplotlib.pyplot as plt
+          import numpy as np
+          from palmerpenguins import load_penguins
+          from shiny import App, render, ui
+
+          app_ui = ui.page_fluid(
+              ui.input_slider("n", "Number of bins", 0, 100, 20),
+              ui.output_plot("plot"),  #<<
+          )
+
+
+          def server(input, output, session):
+              @render.plot(alt="A histogram")  #<<
+              def plot():
+                  df = load_penguins()
+                  mass = df["body_mass_g"]
+
+                  fig, ax = plt.subplots()  #<<
+                  ax.hist(mass, input.n(), density=True)
+                  ax.set_title("Palmer Penguin Masses")
+                  ax.set_xlabel("Mass (g)")
+                  ax.set_ylabel("Density")
+
+                  return fig  #<<
+
+
+          app = App(app_ui, server, debug=True)
+
+      relevantfunctions:
+        - title: ui.output_plot
+          href: https://shiny.posit.co/py/api/ui.output_plot.html
+          signature: ui.output_plot(id, width='100%', height='400px', *, inline=False, click=False, dblclick=False, hover=False, brush=False, fill=MISSING)
+        - title: "@render.plot"
+          href: https://shiny.posit.co/py/api/render.plot.html
+          signature: render.plot(_fn=None, *, alt=None, width=MISSING, height=MISSING, **kwargs)
+        - title: matplotlib.pyplot.subplots
+          href: https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots.html
+          signature: matplotlib.pyplot.subplots(nrows=1, ncols=1, *, sharex=False, sharey=False, squeeze=True, width_ratios=None, height_ratios=None, subplot_kw=None, gridspec_kw=None, **fig_kw)
+        - title: matplotlib.axes.Axes.hist
+          href: https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.hist.html
+          signature: Axes.hist(x, bins=None, range=None, density=False, weights=None, cumulative=False, bottom=None, histtype='bar', align='mid', orientation='vertical', rwidth=None, log=False, color=None, label=None, stacked=False, *, data=None, **kwargs)
+
+      details: |
+
+              <p>The Matplotlib is a library that can be used to create plots.</p>
+
+              <p>Follow three steps to display a Matplotlib figure in your app:</p>
+
+              <ol>
+                <li>
+                Add `ui.output_plot()` to the UI of your app to create a div in which to display the figure.
+                Where you call this function will determine where the figure will appear within the layout of the app.
+                </li>
+                <li>
+                Define a function within the `server()` function that creates the figure and
+                is wrapped around with the `@render.plot()` decorator to return the figure.
```

Comment:
I was a little confused about how to use `@render.plot()` here because it is mentioned in two places. I think it makes more sense to take out this mention of `@render.plot()` but keep the one below.

Reply:
I now see that the second mention of `@render.plot()` is a sub-bullet of this one, but I still find it a little confusing.